--- grammar/src/bin/snapshot.rs.orig	2026-04-24 23:55:31.047361779 +0300
+++ grammar/src/bin/snapshot.rs	2026-04-24 23:54:40.349576022 +0300
@@ -1,10 +1,16 @@
 use svg::Document;
-use svg::node::element::{Circle, Definitions, Line, Marker, Path, Style, Text};
+use svg::node::element::{Circle, Definitions, Line, Marker, Path, Rectangle, Style, Text};
 
 fn main() {
     let mut document = Document::new()
         .set("viewBox", (0, 0, 500, 500))
-        .set("xmlns", "http://www.w3.org/2000/svg");
+        .set("xmlns", "http://www.w3.org/2000/svg")
+        .add(
+            Rectangle::new()
+                .set("width", "100%")
+                .set("height", "100%")
+                .set("fill", "white"),
+        );
 
     // ---------- стиль для шрифта ----------
     // Если шрифт не установлен локально, браузер может загрузить его через @font-face,
